arch/riscv/kernel/devicetree.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Enables use of Device Tree for RISC-V peripheral devices.
+ *
+ * Author: Jamey Hicks <jamey.hicks@gmail.com>
+ */
+
+#include <linux/module.h>
+#include <linux/types.h>
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/slab.h>
+#include <linux/device.h>
+#include <linux/io.h>
+#include <linux/irqchip.h>
+#include <linux/of_fdt.h>
+#include <linux/of_platform.h>
+
+int __init devicetree_init(void)
+{
+	resource_size_t initial_dtb = 0x04100000;
```

Comment:
We should avoid hard-coding a physical address for the FDT as well as directly assigning to `initial_boot_params`.  It appears that the usual approach is to pass a pointer to `early_init_dt_verify()`, which might either be provided by the bootloader or statically initialized to `__dtb_start` when using the builtin FDT.

More generally, device tree handling doesn't seem appropriate for a standalone kernel module.  It should probably be part of `setup_arch()`, as in most other architectures, preferably written in a way that respects `CONFIG_OF`.